src/debug/shared/dbgtransportsession.cpp

```diff
@@ -57,6 +57,11 @@ HRESULT DbgTransportSession::Init(DebuggerIPCControlBlock *pDCB, AppDomainEnumer
     // cleanup necessary.
     memset(this, 0, sizeof(*this));
 
+    // Because of the above memset the embeded classes/structs need to be reinitialized especially
+    // the two way pipe; it expects the in/out handles to be -1 instead of 0.
```

Comment:
It is definitely strange and I don’t like it either but I guess I don’t have time (right now) to do the right thing (properly initialize each field).  I wish it was done properly when it was first written.